dbms/src/Common/MiAllocator.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <Common/config.h>
+
+#if !USE_MIMALLOC
+#error "do not include this file until USE_MIMALLOC is set to 1"
+#endif
+
+#include <mimalloc.h>
+#include <cstddef>
+
+namespace DB
+{
+struct MiAllocator
+{
+
+    static void * alloc(size_t size, size_t alignment = 0)
```

Comment:
If you move to .cpp, you don't have to make any changes to 
`dbms/src/Interpreters/Compiler.cpp`

Reply:
But we lose the dead code elimination property if `alignment` is zero which is rather common case

Reply:
Reasonable.